### ISSUE TITLE
[KGCL] New Image Axiom Request for cardiomyocyte

### ISSUE THREAD

**Issue by lubianat** [obophenotype/cell-ontology#2298]
**Request to add a new image axiom with KGCL**

**Wikidata item:** [cardiomyocyte](https://www.wikidata.org/entity/Q1047227)
**CL ID:** http://purl.obolibrary.org/obo/CL_0000746
**Image URL:** http://commons.wikimedia.org/wiki/Special:FilePath/1020%20Cardiac%20Muscle.jpg

```
      Hey ontobot! apply: - create edge http://purl.obolibrary.org/obo/CL_0000746 <http://xmlns.com/foaf/0.1/depicted_by> "http://commons.wikimedia.org/wiki/Special:FilePath/1020%20Cardiac%20Muscle.jpg"^^xsd:anyURI
```

**Additional notes or concerns:**
This request was autogenerated from Wikidata.

**Comment by lubianat** [comment]
@hrshdhgd and @gouttegd here is an attempt at a simple edge change with KGCL, not sure if full URI syntax can be used.

**Comment by hrshdhgd** [comment]
> @hrshdhgd and @gouttegd here is an attempt at a simple edge change with KGCL, not sure if full URI syntax can be used.

URI syntax can used but we recommend using CURIEs for readability. It's user/curator preference really.

**Comment by hrshdhgd** [comment]
I tried


Hey ontobot! apply: 
- create edge http://purl.obolibrary.org/obo/CL_0000746 <http://xmlns.com/foaf/0.1/depicted_by> "http://commons.wikimedia.org/wiki/Special:FilePath/1020%20Cardiac%20Muscle.jpg"^^xsd:anyURI

[on my fork](https://github.com/hrshdhgd/cell-ontology/issues/16) and I get this [error](https://github.com/hrshdhgd/cell-ontology/actions/runs/8061441015/job/22019212002):
```
2024-02-27 08:04:16,027 ERROR org.incenp.obofoundry.kgcl.robot.ApplyCommand - KGCL syntax error: line 1, column 12: no viable alternative at input 'createedge'http://purl.obolibrary.org/obo/CL_0000746''
```

Thoughts @gouttegd ?

**Comment by gouttegd** [comment]
> Perhaps a space missing between create and edge?

No. The root cause is the use of full IRI not enclosed within `<` and `>` characters.

This is something that the Python implementation of KGCL does allow, but my Java implementation does not (purposefully).

KGCL-Java follows the same logic as the one used, for example, in the OWL functional syntax: identifiers are either “naked“ CURIES or angle-bracketed full IRIs. So it’s either `CL:0000746` or `<http://purl.obolibrary.org/obo/CL_0000746>`, but a “naked” full IRI (`http://purl.obolibrary.org/obo/CL_0000746>`) will _not_ be recognised.

Beyond this, note that there are several other problems with that command:

* `http://xmlns.com/foaf/1.0/depicted_by` does not exist. The [correct property](http://xmlns.com/foaf/spec/#term_depiction) is `http://xmlns.com/foaf/1.0/depiction`.
* That property is an _annotation_ property, not an _object_ property. It cannot be used as the predicate of a `create edge` command (the predicate can only be either `rdfs:subClassOf` or an _object_ property).
* Currently, the KGCL DSL spec does not define a syntax for adding an arbitrary annotation. It only has a command for _replacing_ an existing annotation (`change annotation of CL:0000746 with oaf:depiction from "old value" to "new value"`).

**Comment by gouttegd** [comment]
> It cannot be used as the predicate of a create edge command (the predicate can only be either rdfs:subClassOf or an object property).

Upon re-reading the [KGCL definition of “edge”](https://incatools.github.io/kgcl/Edge/), this is actually an error of my part. An edge _can_ use an annotation property as the predicate, provided the object side is itself a node and not a literal (if the object is a literal then this is an annotation, not an edge). So KGCL-Java refusing to create an edge in this case is wrong.

**Comment by lubianat** [comment]
@gouttegd  Unfortunately, though depicted_by does not exist, it is the one used by CL and Uberon. It is a long standing, low priority issue (see e.g. #753) 

So, from the last comment, with the proper <> it might work?

**Comment by lubianat** [comment]
Oh it was actually fixed by @anitacaron apparently! that is great.

**Comment by lubianat** [comment]
So, what could work then is 

Hey ontobot! apply:

create edge <http://purl.obolibrary.org/obo/CL_0000746> <http://xmlns.com/foaf/0.1/depiction> <http://commons.wikimedia.org/wiki/Special:FilePath/1020%20Cardiac%20Muscle.jpg>

**Comment by gouttegd** [comment]
Just because we have been using the wrong property the whole time, does not mean we should keep doing so. And CL has already fixed that on its side, there’s no more `foaf:depicted_by` in CL right now as far as I can tell.

> So, from the last comment, with the proper <> it might work?

No. `create edge <http://purl.obolibrary.org/obo/CL_0000746> <http://xmlns.com/foaf/0.1/depiction> <http://commons.wikimedia.org/wiki/Special:FilePath/1020%20Cardiac%20Muscle.jpg>` _should_ work but will not for now because of a bug in KGCL-Java (which incorrectly refuses to create an edge using an annotation property).

And besides, the way `foaf:depiction` has typically be used so far is with a _literal_ value (in OWL terms: `AnnotationAssertion(foaf:depiction CL:0000746 "http://commons.wikimedia.org/wiki/Special:FilePath/1020%20Cardiac%20Muscle.jpg"^^xsd:anyURI)`), rather than with an _object_ value (in OWL terms: `AnnotationAssertion(foaf:depiction CL:0000746 <http://commons.wikimedia.org/wiki/Special:FilePath/1020%20Cardiac%20Muscle.jpg>)`). KGCL itself does not allow to that for now.

**Comment by gouttegd** [comment]
> No. create edge <http://purl.obolibrary.org/obo/CL_0000746> <http://xmlns.com/foaf/0.1/depiction> <http://commons.wikimedia.org/wiki/Special:FilePath/1020%20Cardiac%20Muscle.jpg> should work but will not for now because of a bug in KGCL-Java (which incorrectly refuses to create an edge using an annotation property).

That particular bug is now fixed. With the next release of KGCL-Java,

```
create edge CL:0000746 foaf:depiction <http://commons.wikimedia.org/wiki/Special:FilePath/1020%20Cardiac%20Muscle.jpg>
```

_will_ work.

But this does not change the fact that we would really need to create here is an _annotation_, not an _edge_, and KGCL itself needs to be amended to allow that.
